src/cassiopeia/critique/compare.py

```diff
@@ -8,110 +8,285 @@
+    Trees must have identical leaf sets.
 
     Args:
-        tree1: Input CassiopeiaTree
-        tree2: CassiopeiaTree to be compared to the first tree.
-        number_of_trials: Number of triplets to sample at each depth
-        min_triplets_at_depth: The minimum number of triplets needed with LCA
-            at a depth for that depth to be included
+        tree1: First tree (`CassiopeiaTree`, `nx.DiGraph`, or str key into `tdata.obst`).
+        tree2: Second tree of the same type as `tree1`.
+        tdata: Required only when `tree1` and `tree2` are string keys; must provide `.obst`.
+        number_of_trials: Number of triplets to sample at each depth.
+        min_triplets_at_depth: Minimum number of triplets with LCA at a depth for that
+            depth to be included.
 
     Returns
     -------
-        A tuple of four dictionaries storing triplet statistics at each depth:
+        A tuple of four dicts keyed by depth:
             all_triplets_correct: Total proportion of triplets correct.
             resolvable_triplets_correct: Proportion correct among resolvable triplets.
             unresolved_triplets_correct: Proportion correct among unresolvable triplets.
-            proportion_resolvable: Proportion of triplets that are resolvable at each depth.
+            proportion_unresolvable: Proportion of triplets that are unresolvable at each depth.
     """
-    # keep dictionary of triplets correct
-    all_triplets_correct = defaultdict(int)
-    unresolved_triplets_correct = defaultdict(int)
-    resolvable_triplets_correct = defaultdict(int)
-    proportion_unresolvable = defaultdict(int)
+    if type(tree1) is not type(tree2):
+        raise TypeError("tree1 and tree2 must be the same type. ")
+
+    if isinstance(tree1, CassiopeiaTree):
+        G1 = to_nx_tree(copy.deepcopy(tree1))
+        G2 = to_nx_tree(copy.deepcopy(tree2))
+
+    elif isinstance(tree1, str):
+        if tdata is None:
+            raise ValueError("When tree1 and tree2 are strings, tdata must be provided")
+        if not hasattr(tdata, "obst") or tdata.obst is None or len(tdata.obst) == 0:
+            raise ValueError("tdata does not have an 'obst' attribute")
+        if tree1 not in tdata.obst or tree2 not in tdata.obst:
+            raise ValueError(
+                f"Tree keys must exist in tdata.obst. Missing: {[k for k in [tree1, tree2] if k not in tdata.obst]}"
+            )
+
+        # tdata key triplets correct
+        raw1 = tdata.obst[tree1]
+        raw2 = tdata.obst[tree2]
+        G1 = to_nx_tree(copy.deepcopy(raw1))
+        G2 = to_nx_tree(copy.deepcopy(raw2))
+
+    elif isinstance(tree1, nx.DiGraph):
+        # nx.DiGraph triplets correct
+        G1 = to_nx_tree(copy.deepcopy(tree1))
+        G2 = to_nx_tree(copy.deepcopy(tree2))
+
+    else:
+        raise TypeError("Unsupported input type. Expected CassiopeiaTree, str (key into tdata.obst), or nx.DiGraph.")
```

Comment:
Ditto comment above - if we throw an error in a certain situation, this should be made clear in the docstring.